apps/tx-builder/src/utils.test.ts

```diff
@@ -0,0 +1,1506 @@
+          ])
+        })
+      })
+
+      describe('uint[] & uint[size]', () => {
+        it('parse an array of numbers to array of strings', () => {
+          expect(parseInputValue('uint[]', '[1]')).toEqual([toBN('1').toString(10, 256)])
+          expect(parseInputValue('uint[2]', '[1]')).toEqual([toBN('1').toString(10, 256)])
+
+          expect(parseInputValue('uint120[]', '[1]')).toEqual([toBN('1').toString(10, 120)])
+          expect(parseInputValue('uint120[2]', '[1]')).toEqual([toBN('1').toString(10, 120)])
+        })
+
+        it('parse an array of numbers as strings to array of strings', () => {
+          expect(parseInputValue('uint[]', '["1"]')).toEqual([toBN('1').toString(10, 256)])
+          expect(parseInputValue('uint[2]', '["1"]')).toEqual([toBN('1').toString(10, 256)])
+
+          expect(parseInputValue('uint120[]', '["1"]')).toEqual([toBN('1').toString(10, 120)])
+          expect(parseInputValue('uint120[2]', '["1"]')).toEqual([toBN('1').toString(10, 120)])
+        })
+
+        it('parse an array of long numbers to array of strings', () => {
+          expect(parseInputValue('uint[]', '[6426191757410075707]')).toEqual([
+            toBN('6426191757410075707').toString(10, 256),
+          ])
+          expect(parseInputValue('uint[4]', '[6426191757410075707]')).toEqual([
+            toBN('6426191757410075707').toString(10, 256),
+          ])
+
+          expect(parseInputValue('uint120[]', '[6426191757410075707]')).toEqual([
+            toBN('6426191757410075707').toString(10, 120),
+          ])
+          expect(parseInputValue('uint120[4]', '[6426191757410075707]')).toEqual([
+            toBN('6426191757410075707').toString(10, 120),
+          ])
+        })
+
+        it('parse an array of long numbers as strings to array of strings', () => {
+          expect(parseInputValue('uint[]', '["6426191757410075707"]')).toEqual([
+            toBN('6426191757410075707').toString(10, 256),
+          ])
+          expect(parseInputValue('uint[3]', '["6426191757410075707"]')).toEqual([
+            toBN('6426191757410075707').toString(10, 256),
+          ])
+
+          expect(parseInputValue('uint120[]', '["6426191757410075707"]')).toEqual([
+            toBN('6426191757410075707').toString(10, 120),
+          ])
+          expect(parseInputValue('uint120[3]', '["6426191757410075707"]')).toEqual([
+            toBN('6426191757410075707').toString(10, 120),
+          ])
+        })
+
+        it('parse an empty array as valid value', () => {
+          expect(parseInputValue('uint[]', '[]')).toEqual([])
+          expect(parseInputValue('uint[2]', '[]')).toEqual([])
+
+          expect(parseInputValue('uint120[]', '[]')).toEqual([])
+          expect(parseInputValue('uint120[2]', '[]')).toEqual([])
+        })
+
+        it('parse an invalid array throws a SyntaxError', () => {
+          expect(() => parseInputValue('uint[]', 'invalid_array_value')).toThrow(SyntaxError)
```

Comment:
I selected the `SyntaxError` because `JSON.parse()` throws a `SyntaxError` exception if the string to parse is not valid JSON.

But we can change it, what error can we throw ?

Reply:
perhaps a custom error like `SoliditySyntaxError` or something like that to highlight that the error is not thrown by the JS engine but by us

Reply:
Done! I added a custom SoliditySyntaxError to address it!